server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java

```diff
@@ -479,7 +479,7 @@ private String login(Model model, Principal principal, List<String> excludedProm
                     return "idp_discovery/account_chooser";
                 }
 
-                if (!discoveryPerformed && !loginHintParam.isPresent()) {
+                if (!discoveryPerformed && !loginHintParam.isPresent() && combinedIdps.size() > 0) {
```

Comment:
Better to consistently use `!<collection>.isEmpty()` instead of `<collection>.size() > 0` since the latter can be O(N) for some data structures but the former never is.